### PR TITLE
Add support for cubic lattice embeddings on zephyr to make_origin_embedding

### PR DESCRIPTION
**Pull request by jackraymond**
Creates chain length two embeddings on a target zephyr graph, permissive of node defects, with optional permissiveness of edge defects. Matches the template for other make_origin_embedding routines.

### CODE REVIEW

**Comment by jackraymond** [comment]
Requested by @pau557

File: hybrid/decomposers.py [hunk 1]
```diff
@@ -1161,25 +1161,41 @@ def make_origin_embeddings(qpu_sampler=None, lattice_type=None,
...
+            dimensions = [L, L, 4*t]
+            # A suitable, but not unique, scheme for L ~ 16 at chain length 2.
+            def zephyr_chain(x, y, z):
+                return (vec_to_lin((0, 2*y + (z//t)%2, z%t, z//(2*t), x)),
+                        vec_to_lin((1, 2*x + (z//t)%2, z%t, z//(2*t), y)))
+            origin_embedding = {(x, y, z): zephyr_chain(x,y,z)
+                                for x in range(dimensions[0])
+                                for y in range(dimensions[1])
+                                for z in range(dimensions[2])
+                                if target.has_edge(*zephyr_chain(x, y, z))}
```
[the remark this answers is not in the document]
Comment (randomir) [reply hybrid/decomposers.py:1177]: Another code simplification you could do (same performance, but less clarity gratis :laughing:), with `product` from `itertools`:
```suggestion
            origin_embedding = {e: c
                                for e in product(*map(range, dimensions))
                                if target.has_edge(*(c := zephyr_chain(*e)))}
```

File: tests/test_decomposers.py [hunk 2]
```diff
@@ -719,8 +720,9 @@ def test_all_embedding_shapes(self):
             elif qpu_top == 'pegasus':
                 lattice_types = ['cubic', 'kings', qpu_top, None]
             elif qpu_top == 'zephyr':
-                lattice_types = ['kings']
-            
+                lattice_types = ['cubic', 'kings']
+            else:
+                raise ValueError('Uknown qpu topology')
```
Comment (randomir) [inline tests/test_decomposers.py:725]: Typo.

**Review by randomir — APPROVED** [review]
LGTM!

File: tests/test_decomposers.py (hunk 2 above)
Comment (jackraymond) [inline tests/test_decomposers.py:725]:
```suggestion
                raise ValueError('Unknown qpu topology')
```

File: hybrid/decomposers.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jackraymond) [reply hybrid/decomposers.py]: I implemented this simplification as suggested, I also simplified the dereferencing in other parts of the same file following this pattern.

**Review by randomir — APPROVED** [review]
Looks great, thanks Jack!

**Comment by mahdiehmalekian** [comment]
The embeddings for 'cubic' on zephyr looks great; my only point is that it embeds `m x m x 3t` (not `4t`) cubic lattices. Even though the  `zephyr_chain` gives the formula of chains for `z= 0, ..., 4t-1`, the issue with the formula for `z= 2t, ..., 3t-1`  is that they are not  chains, meaning there is no edge in the Zephyr graph between the two nodes in the chain. Thanks to the current line 1173 `if target.has_edge(*(c := zephyr_chain(*coord)))` the given embedding is correct (only it misses `z= 2t, ..., 3t-1` and so works for `m x m x 3t`).

**Pull request merged**